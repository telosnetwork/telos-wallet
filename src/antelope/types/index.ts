// interfaces for antelope
export * from 'src/antelope/types/ABIv1';
export * from 'src/antelope/types/Actions';
export * from 'src/antelope/types/Api';
export * from 'src/antelope/types/ChainInfo';
export * from 'src/antelope/types/ChainSettings';
export * from 'src/antelope/types/EvmContractData';
export * from 'src/antelope/types/EvmLog';
export * from 'src/antelope/types/EvmBlockData';
export * from 'src/antelope/types/EvmTransaction';
export * from 'src/antelope/types/ExceptionError';
export * from 'src/antelope/types/Filters';
export * from 'src/antelope/types/PriceData';
export * from 'src/antelope/types/Proposals';
export * from 'src/antelope/types/Producers';
export * from 'src/antelope/types/KeyAccounts';
export * from 'src/antelope/types/Label';
export * from 'src/antelope/types/Theme';
export * from 'src/antelope/types/TransactionV1';


// classes for antelope
export * from 'src/antelope/types/AntelopeError';

// interfaces for antelope evm-abi
export * from 'src/antelope/stores/utils/abi';
