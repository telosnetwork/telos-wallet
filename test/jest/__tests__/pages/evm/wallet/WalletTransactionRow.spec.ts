import { shallowMount } from '@vue/test-utils';
import { stubWithSlot } from 'test/jest/testing-helpers';

const fakeExplorerUrl = 'example.com';
const locale = 'en-US';
const currency = 'USD';

const antelopeMock = {
    useChainStore: () => ({
        currentChain: {
            settings: {
                getExplorerUrl: () => fakeExplorerUrl,
                getSystemToken: () => ({
                    symbol: 'TLOS',
                }),
            },
        },
    }),
    useUserStore: () => ({
        fiatLocale: locale,
        fiatCurrency: currency,
    }),
};

import WalletTransactionRow from 'pages/evm/wallet/WalletTransactionRow.vue';
import { ShapedTransactionRow } from 'src/antelope/types';

jest.mock('src/antelope', () => antelopeMock);


describe('WalletTransactionRow.vue', () => {
    const fakeDateSeconds = 1681775186;

    beforeAll(() => {
        // used so that timestamp text is the same for each test run, not depending on the day it is run
        jest.useFakeTimers('modern').setSystemTime(new Date(fakeDateSeconds * 1000));
    });

    afterAll(() => {
        jest.useRealTimers();
    });

    const defaultTxData: ShapedTransactionRow = {
        id: '0x'.concat('1'.repeat(40)),
        epoch: fakeDateSeconds,
        actionName: '',
        from: '0x'.concat('1'.repeat(40)),
        fromPrettyName: '',
        to: '0x'.concat('2'.repeat(40)),
        toPrettyName: '',
        valuesIn: [],
        valuesOut: [],
        nftsIn: [],
        nftsOut: [],
        gasUsed: 0.15,
        gasFiatValue: 0.03,
    };

    function getMountOptions(tx: ShapedTransactionRow, isMobile: boolean) {
        return {
            global: {
                mocks: {
                    $q: {
                        screen: {
                            lt: {
                                sm: isMobile,
                            },
                            gt: {
                                xs: !isMobile,
                            },
                        },
                    },
                },
                stubs: {
                    'q-icon': stubWithSlot('q-icon'),
                },
            },
            props: {
                transaction: tx,
            },
        };
    }

    function runExpects(tx: ShapedTransactionRow) {
        const mobileWrapper = shallowMount(WalletTransactionRow, getMountOptions(tx, true));
        const desktopWrapper = shallowMount(WalletTransactionRow, getMountOptions(tx, true));

        expect(mobileWrapper.element).toMatchSnapshot();
        expect(desktopWrapper.element).toMatchSnapshot();
    }


    it('should have the right name', () => {
        expect(WalletTransactionRow.name).toBe('WalletTransactionRow');
    });

    describe('row should be rendered correctly for', () => {
        test('failed transactions', () => {
            const tx = {
                ...defaultTxData,
                failed: true,
            };

            runExpects(tx);
        });

        test('addresses with an alias', () => {
            const sendTx = {
                ...defaultTxData,
                actionName: 'send',
                toPrettyName: 'SomeContract',
            };

            const otherTx= {
                ...defaultTxData,
                actionName: 'receive',
                fromPrettyName: 'AContract',
            };

            runExpects(sendTx);
            runExpects(otherTx);
        });

        test('transfers with no fiat price', () => {
            const tx = {
                ...defaultTxData,
                actionName: 'swap',
                toPrettyName: 'SomeContract',
                valuesIn: [{
                    amount: 100.54121,
                    symbol: 'NERD',
                }, {
                    amount: 357542547.674235,
                    symbol: 'SHIB',
                }, {
                    amount: 1307.45,
                    symbol: 'TLOS',
                    fiatValue: 54.45123513,
                }],
                valuesOut: [{
                    amount: 100,
                    symbol: 'sTLOS',
                    fiatValue: 100,
                }],
            };

            runExpects(tx);
        });

        test('transactions with an action name other than send/receive/swap', () => {
            const tx = {
                ...defaultTxData,
                actionName: 'approve',
            };

            runExpects(tx);
        });

        test('transactions with no known method name', () => {
            const tx = {
                ...defaultTxData,
            };

            runExpects(tx);
        });
    });
});
