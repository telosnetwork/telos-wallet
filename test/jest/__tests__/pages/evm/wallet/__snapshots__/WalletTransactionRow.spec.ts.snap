// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletTransactionRow.vue row should be rendered correctly for addresses with an alias 1`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        
        <span
          class="c-transaction-row__action-description"
        >
          evm_wallet.sent  
        </span>
        <span
          class="c-transaction-row__interaction-nowrap"
        >
          global.to 
          <external-link-stub
            purpose="evm_wallet.aria_link_to_address"
            text="SomeContract"
            url="example.com/address/0x2222222222222222222222222222222222222222"
          />
        </span>
        
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    
    
    
    
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;

exports[`WalletTransactionRow.vue row should be rendered correctly for addresses with an alias 2`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        
        <span
          class="c-transaction-row__action-description"
        >
          evm_wallet.sent  
        </span>
        <span
          class="c-transaction-row__interaction-nowrap"
        >
          global.to 
          <external-link-stub
            purpose="evm_wallet.aria_link_to_address"
            text="SomeContract"
            url="example.com/address/0x2222222222222222222222222222222222222222"
          />
        </span>
        
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    
    
    
    
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;

exports[`WalletTransactionRow.vue row should be rendered correctly for addresses with an alias 3`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon c-transaction-row__interaction-icon--rotated"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        
        <span
          class="c-transaction-row__action-description"
        >
          evm_wallet.received  
        </span>
        <span
          class="c-transaction-row__interaction-nowrap"
        >
          global.from 
          <external-link-stub
            purpose="evm_wallet.aria_link_to_address"
            text="AContract"
            url="example.com/address/0x1111111111111111111111111111111111111111"
          />
        </span>
        
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    
    
    
    
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;

exports[`WalletTransactionRow.vue row should be rendered correctly for addresses with an alias 4`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon c-transaction-row__interaction-icon--rotated"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        
        <span
          class="c-transaction-row__action-description"
        >
          evm_wallet.received  
        </span>
        <span
          class="c-transaction-row__interaction-nowrap"
        >
          global.from 
          <external-link-stub
            purpose="evm_wallet.aria_link_to_address"
            text="AContract"
            url="example.com/address/0x1111111111111111111111111111111111111111"
          />
        </span>
        
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    
    
    
    
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;

exports[`WalletTransactionRow.vue row should be rendered correctly for failed transactions 1`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon c-transaction-row__interaction-icon--red"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        <span
          class="c-transaction-row__action-name"
        >
          evm_wallet.failed_contract_interaction
        </span>
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    
    
    
    
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;

exports[`WalletTransactionRow.vue row should be rendered correctly for failed transactions 2`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon c-transaction-row__interaction-icon--red"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        <span
          class="c-transaction-row__action-name"
        >
          evm_wallet.failed_contract_interaction
        </span>
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    
    
    
    
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;

exports[`WalletTransactionRow.vue row should be rendered correctly for transactions with an action name other than send/receive/swap 1`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        <span
          class="c-transaction-row__action-name"
        >
          approve
        </span>
      </div>
      <div
        class="c-transaction-row__secondary-interaction-text"
      >
        <external-link-stub
          purpose="evm_wallet.aria_link_to_address"
          text="0x2222222222222222222222222222222222222222"
          url="example.com/address/0x2222222222222222222222222222222222222222"
        />
      </div>
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    
    
    
    
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;

exports[`WalletTransactionRow.vue row should be rendered correctly for transactions with an action name other than send/receive/swap 2`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        <span
          class="c-transaction-row__action-name"
        >
          approve
        </span>
      </div>
      <div
        class="c-transaction-row__secondary-interaction-text"
      >
        <external-link-stub
          purpose="evm_wallet.aria_link_to_address"
          text="0x2222222222222222222222222222222222222222"
          url="example.com/address/0x2222222222222222222222222222222222222222"
        />
      </div>
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    
    
    
    
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;

exports[`WalletTransactionRow.vue row should be rendered correctly for transactions with no known method name 1`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        <span
          class="c-transaction-row__action-name"
        >
          evm_wallet.contract_interaction
        </span>
      </div>
      <div
        class="c-transaction-row__secondary-interaction-text"
      >
        <external-link-stub
          purpose="evm_wallet.aria_link_to_address"
          text="0x2222222222222222222222222222222222222222"
          url="example.com/address/0x2222222222222222222222222222222222222222"
        />
      </div>
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    
    
    
    
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;

exports[`WalletTransactionRow.vue row should be rendered correctly for transactions with no known method name 2`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        <span
          class="c-transaction-row__action-name"
        >
          evm_wallet.contract_interaction
        </span>
      </div>
      <div
        class="c-transaction-row__secondary-interaction-text"
      >
        <external-link-stub
          purpose="evm_wallet.aria_link_to_address"
          text="0x2222222222222222222222222222222222222222"
          url="example.com/address/0x2222222222222222222222222222222222222222"
        />
      </div>
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    
    
    
    
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;

exports[`WalletTransactionRow.vue row should be rendered correctly for transfers with no fiat price 1`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        
        <span
          class="c-transaction-row__action-description"
        >
          evm_wallet.swapped  
        </span>
        <span
          class="c-transaction-row__interaction-nowrap"
        >
          global.with 
          <external-link-stub
            purpose="evm_wallet.aria_link_to_address"
            text="SomeContract"
            url="example.com/address/0x2222222222222222222222222222222222222222"
          />
        </span>
        
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    <div
      class="c-transaction-row__value-container"
    >
      <div
        class="c-transaction-row__value c-transaction-row__value--out"
      >
        -100.0000 sTLOS 
        <!--v-if-->
      </div>
      <span
        class="c-transaction-row__value c-transaction-row__value--out c-transaction-row__value--small"
      >
        -100.00 USD
      </span>
    </div>
    
    
    
    
    <div
      class="c-transaction-row__value-container"
    >
      <div
        class="c-transaction-row__value c-transaction-row__value--in"
      >
         +100.5412 NERD 
        <tool-tip-stub
          hideicon="false"
          warnings="evm_wallet.no_fiat_value"
        />
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__value-container"
    >
      <div
        class="c-transaction-row__value c-transaction-row__value--in"
      >
         +357,542,547.6742 SHIB 
        <tool-tip-stub
          hideicon="false"
          warnings="evm_wallet.no_fiat_value"
        />
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__value-container"
    >
      <div
        class="c-transaction-row__value c-transaction-row__value--in"
      >
         +1,307.4500 TLOS 
        <!--v-if-->
      </div>
      <span
        class="c-transaction-row__value c-transaction-row__value--in c-transaction-row__value--small"
      >
         +54.45 USD
      </span>
    </div>
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;

exports[`WalletTransactionRow.vue row should be rendered correctly for transfers with no fiat price 2`] = `
<div
  class="c-transaction-row"
>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--first"
  >
    <div
      class="c-transaction-row__interaction-icon-container"
    >
      <inline-svg-stub
        aria-hidden="true"
        class="c-transaction-row__interaction-icon"
        height="12"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
    </div>
    <div
      class="c-transaction-row__interaction-text-container"
    >
      <div
        class="c-transaction-row__primary-interaction-text"
      >
        
        <span
          class="c-transaction-row__action-description"
        >
          evm_wallet.swapped  
        </span>
        <span
          class="c-transaction-row__interaction-nowrap"
        >
          global.with 
          <external-link-stub
            purpose="evm_wallet.aria_link_to_address"
            text="SomeContract"
            url="example.com/address/0x2222222222222222222222222222222222222222"
          />
        </span>
        
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__timestamp"
    >
      <tool-tip-stub
        hideicon="true"
        text="Apr 17, 2023 11:46:26 PM (UTC +00:00)"
      />
      <!--v-if-->
    </div>
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--second"
  >
    
    <div
      class="c-transaction-row__value-container"
    >
      <div
        class="c-transaction-row__value c-transaction-row__value--out"
      >
        -100.0000 sTLOS 
        <!--v-if-->
      </div>
      <span
        class="c-transaction-row__value c-transaction-row__value--out c-transaction-row__value--small"
      >
        -100.00 USD
      </span>
    </div>
    
    
    
    
    <div
      class="c-transaction-row__value-container"
    >
      <div
        class="c-transaction-row__value c-transaction-row__value--in"
      >
         +100.5412 NERD 
        <tool-tip-stub
          hideicon="false"
          warnings="evm_wallet.no_fiat_value"
        />
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__value-container"
    >
      <div
        class="c-transaction-row__value c-transaction-row__value--in"
      >
         +357,542,547.6742 SHIB 
        <tool-tip-stub
          hideicon="false"
          warnings="evm_wallet.no_fiat_value"
        />
      </div>
      <!--v-if-->
    </div>
    <div
      class="c-transaction-row__value-container"
    >
      <div
        class="c-transaction-row__value c-transaction-row__value--in"
      >
         +1,307.4500 TLOS 
        <!--v-if-->
      </div>
      <span
        class="c-transaction-row__value c-transaction-row__value--in c-transaction-row__value--small"
      >
         +54.45 USD
      </span>
    </div>
    
    
    
  </div>
  <div
    class="c-transaction-row__info-container c-transaction-row__info-container--third"
  >
    <div
      class="c-transaction-row__gas-icon-container"
    >
      <q-icon-stub
        class="c-transaction-row__gas-icon"
        name="o_local_gas_station"
        size="xs"
      >
        
        
      </q-icon-stub>
    </div>
    <div
      class="c-transaction-row__gas-text"
    >
      <span>
        0.1500 TLOS
      </span>
      <span>
        $0.03 USD
      </span>
    </div>
    <div
      class="c-transaction-row__mobile-tx-link"
    >
      <external-link-stub
        purpose="evm_wallet.aria_link_to_transaction"
        text="global.more_info"
        url="example.com/tx/0x1111111111111111111111111111111111111111"
      />
    </div>
  </div>
</div>
`;
